modules/tinymce/src/core/main/ts/delete/TableDelete.ts

```diff
@@ -99,11 +112,30 @@ const emptyMultiTableCells = (
   endTableCells: SugarElement<HTMLTableCellElement>[],
   betweenRng: Range
 ): boolean => {
-  cleanCells(startTableCells.concat(endTableCells));
+  const rng = editor.selection.getRng();
+
+  const startCell = startTableCells[0];
+  const endCell = endTableCells[endTableCells.length - 1];
+
+  deleteContentInsideCell(startCell, rng, true);
+  deleteContentInsideCell(endCell, rng, false);
+
+  // The cursor is always collapsed back into the start cell, so we never need to clean it
+  const startTableCellsToClean = startTableCells.slice(1);
+
+  let endTableCellsToClean = endTableCells;
+  const lastCell = endTableCells[endTableCells.length - 1];
+  // Only clean empty cells, and the last cell has the potential to still have content
+  if (!editor.dom.isEmpty(lastCell.dom)) {
+    endTableCellsToClean = endTableCells.slice(0, endTableCells.length - 1);
+  }
+
+  cleanCells(startTableCellsToClean.concat(endTableCellsToClean));
   // Delete all content in between the start table and end table
   betweenRng.deleteContents();
-  // Set the cursor back to the start of the original selection
-  setCursorInCell(editor, startTableCells[0]);
+
+  // This will collapse the selection into the cell of the start table
+  editor.selection.collapse(true);
```

Comment:
How do we know for certain the selection is in the start of the table? That is why we always manually set it previously, as it's relying on the native selection being moved/retained as you delete content, which can be very brittle. I'd much prefer us to be manually setting this, both here and in `emptySingleTableCells`.

Reply:
This function is only called when the selection start and end are in different tables. We can manually store the start of the selection before we do any processing and then set it again?

Part of the problem with `emptySingleTableCells` was that it was snapping the selection back into the cell instead of allowing it to leave the table when the start of the selection wasn't in the table previously

Reply:
We need to continue this discussion because even though I've gone back to similar logic as before, it is still collapsing